### PR TITLE
Graph mapping sanity

### PR DESCRIPTION
**Pull request by dkfellows**
Downstream of SpiNNakerManchester/PACMAN#256

### CODE REVIEW

File: fec_integration_tests/utilities/notification_protocol/test_start_resume_notification_protocol.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 from spinnman.messages.eieio.command_messages import EIEIOCommandMessage
 from spinnman.constants import EIEIO_COMMAND_IDS
 from spinn_front_end_common.utilities.notification_protocol import (
-    NotificationProtocol)
+    NotificationProtocolImpl)
```
Comment (Christian-B) [inline fec_integration_tests/utilities/notification_protocol/test_start_resume_notification_protocol.py]: See https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/pull/553

**Comment by Christian-B** [comment]
After reverting NotifiactionPrototcolImpl rename

Adding https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/pull/553 and https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/pull/554 I intended all remaining changes to be actual GraphMapper ones

File: spinn_front_end_common/interface/interface_functions/insert_edges_to_live_packet_gatherers.py [hunk 2]
```diff
@@ -23,192 +23,177 @@
...
-        :param n_keys_map: map between partition and n keys
-        :return: the application edge for this vertex and LPG
-        :rtype: ~pacman.model.graph.application.ApplicationEdge
+        if not app_vertex.machine_vertices or not p_ids:
+            return
+
+        # flag for ensuring we don't add the edge to the app
+        # graph many times
+        app_edges = dict()
+        m_graph = app_graph.machine_graph
```
Comment (alan-stokes) [inline spinn_front_end_common/interface/interface_functions/insert_edges_to_live_packet_gatherers.py:114]: if you instead hand over the machine graph as a param to this method. this call becomes compeltely redundant

**Comment by dkfellows** [comment]
Do #617 before this.

File: spinn_front_end_common/utility_models/live_packet_gather.py [hunk 3]
```diff
@@ -44,10 +44,12 @@ def __init__(self, lpg_params, constraints=None):
 
     @overrides(ApplicationVertex.create_machine_vertex)
     def create_machine_vertex(
-            self, vertex_slice, resources_required,  # @UnusedVariable
+            self, vertex_slice, resources_required,
             label=None, constraints=None):
+        assert(vertex_slice is None)
+        assert(resources_required is None)
```
Comment (dkfellows) [inline spinn_front_end_common/utility_models/live_packet_gather.py:50]: @Christian-B Why did you put these `assert`s in? The general contract for `create_machine_vertex` allows these parameters to be other things (though in general we ought to know the slice and resources for the LPG prior to the call being made as LPGs are pretty simple).

This is causing a failure in `fec_integration_tests.interface.interface_functions.test_front_end_common_insert_edges_to_live_packet_gatherers.TestInsertLPGEdges`

**Pull request merged**